commands/build.go

```diff
@@ -643,6 +653,81 @@ func parsePrintFunc(str string) (*build.PrintFunc, error) {
 	return f, nil
 }
 
+func addGitProvenance(labels []string, contextPath string, dockerfilePath string) ([]string, error) {
+	v, ok := os.LookupEnv("BUILDX_GIT_LABELS")
+	if !ok || contextPath == "" {
+		return labels, nil
+	}
+
+	if len(labels) == 0 {
+		labels = make([]string, 0)
+	}
+
+	// figure out in which directory the git command needs to run
+	var wd string
+	if filepath.IsAbs(contextPath) {
+		wd = contextPath
+	} else {
+		cwd, _ := os.Getwd()
+		wd, _ = filepath.Abs(filepath.Join(cwd, contextPath))
+	}
+
+	// obtain Git sha of current HEAD
+	cmd := exec.Command("git", "rev-parse", "HEAD")
+	cmd.Dir = wd
+	sha, err := cmd.Output()
+	if err != nil {
+		return nil, fmt.Errorf("error obtaining git head: %w", err)
```

Comment:
In the build repos we only use `pkg/errors` as stdlib does not preserve stacktraces.

Reply:
Good point!